storage/filesystem.go

```diff
@@ -156,7 +157,7 @@ func (f *Filesystem) WriteMetadata(volumeID string, meta metadata.Metadata) erro
 }
 
 func (f *Filesystem) RegisterMetadata(meta metadata.Metadata) (bool, error) {
```

Comment:
I think we also need to make a corresponding change to the 'in memory' storage implementation too (which is used during integration testing).

Adding to that, how can we write tests to verify this new functionality?

Reply:
Thanks, I have updated the inmemory store.

Any advise on how to test this functionality? I struggled to test this in `filesystem_test.go` for example.